sycl/test-e2e/Graph/Inputs/multiple_kernel_bundles.cpp

```diff
@@ -0,0 +1,119 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Tests using a bundle in a graph.
+// Checks the PI call trace to ensure that the bundle kernel of the single task
+// is used.
+
+#include "../graph_common.hpp"
+
+class Kernel1Name;
+class Kernel2Name;
+
+int main() {
+  using T = unsigned short;
+
+  const sycl::device Dev{sycl::default_selector_v};
+  const sycl::device Dev2{sycl::default_selector_v};
+
+  const sycl::context Ctx{Dev};
+  const sycl::context Ctx2{Dev2};
+
+  queue Queue{Ctx,
+              Dev,
+              {sycl::ext::intel::property::queue::no_immediate_command_list{}}};
+
+#ifndef GRAPH_E2E_EXPLICIT
```

Comment:
Could you put a code comment why we need this `#ifndef` only for record & replay

Reply:
I added a comment and moved the code to make it clearer.